### DISCUSSION TITLE
Wallbox Modbus-Register von Leistungsaufnahme in Decimal als Zähler

### DISCUSSION THREAD

**Discussion by Lausbua** [camueller/SmartApplianceEnabler#595] [Installation und Betrieb]
Hallo Zusammen,

ich habe eine Wallbox Wallbe Eco 2.0s erfolgreich in Betrieb genommen und diese mit einem Modbus-Zähler nachgerüstet. Der Modbus-Zähler wird von der Wallbox ausgelesen und die Werte werden von der Wallbox in die Modbus-Register der Wallbox geschrieben.

Diese Werte sind Decimal. Unter Modbus-Register 120 (EM active power - Wirkleistung) wird die Leistungsaufnahme geschrieben.

![image](https://github.com/camueller/SmartApplianceEnabler/assets/3309871/c93d587c-b4db-4dc6-86b4-cd80b37306b1)

Wie bringe ich diesen Wert in den SAE?
Der ausgelesene Wert bräuchte dazu einen Korrekturfaktor von 0.1

Mit dieser Einstellung wird der Wert ausgelesen aber kein Korrekturfaktor berechnet.

![image](https://github.com/camueller/SmartApplianceEnabler/assets/3309871/463aee37-e9bf-4c77-8536-01c2019c6fee)

Die Wallbox hat einen Standbyverbrauch von 1,9W. Anbei der ausgelesene Zählerwert und die danach nicht durchgeführte Korrektur.

![image](https://github.com/camueller/SmartApplianceEnabler/assets/3309871/e3b6d670-51c2-48e1-97ee-d76440a72647)

Currentpower 19W obwohl 1,9W.

Wäre es Möglich das ein Wert-Typ Decimal auswählbar wird?

Gruß,
Daniel

**Comment by camueller** [comment]
Die ausgelesenen und korrigierten Werte findest Du im Log. Allerdings macht es keinen Sinn, die Standby-Leistungsaufnahme zu analysieren, weil der Home Manager nur Verbräuche berücksichtigt, wenn das Gerät eingeschaltet ist bzw. lädt.

**Comment by Lausbua** [comment]
Heute erster Test bei reichlich Überschuss. Es funktioniert.

Ich habe in der Wallbox den Korrekturfaktor hinterlegen können damit im Modbusregister die aktuell bezogene Leistung in W steht.
